src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/DefaultRazorFormattingServiceTest.cs

```diff
@@ -543,5 +543,57 @@ public async Task OnTypeCloseAngle_OutsideRazorBlock_DoesNotCloseTextTag()
 ",
 character: ">");
         }
+
+        [Fact]
+        public async Task OnTypeStar_ClosesRazorComment()
+        {
+            await RunFormatOnTypeTestAsync(
+input: @"
+@|
+",
+expected: $@"
+@* {LanguageServerConstants.CursorPlaceholderString} *@
+",
+character: "*");
+        }
+
+        [Fact]
+        public async Task OnTypeStar_InsideRazorComment_Noops()
+        {
+            await RunFormatOnTypeTestAsync(
+input: @"
+@* @| *@
```

Comment:
Do the following inputs work:
- `@*| *@`
- `|@**@`

Reply:
Yup